src/foam/core/types.js

```diff
@@ -431,7 +431,15 @@ foam.CLASS({
   name: 'EMail',
   extends: 'String',
   // FUTURE: verify
-  label: 'Email address'
+  label: 'Email address',
+  properties: [
+    [
+      'adapt',
+      function(_, v) {
+        return v.toLowerCase().trim();
```

Comment:
I'm pretty sure we (actually Kate) added `trim` as a Property property on `foam.core.String` a while back. So you might just be able to do something like this:

```JavaScript
{
  name: 'EMail',
  /* ... */
  trim: true
}
```